servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancerFactory.java

```diff
@@ -95,13 +121,89 @@ public Builder() {
             return this;
         }
 
+        /**
+         * This {@link LoadBalancer} may monitor hosts to which connection establishment has failed
+         * using health checks that run in the background. The health check tries to establish a new connection
+         * and if it succeeds, the host is returned to the load balancing pool. As long as the connection
+         * establishment fails, the host is not considered for opening new connections for processed requests.
+         * <p>
+         * {@link #healthCheckFailedConnectionsThreshold(int)} can be used to disable this mechanism and always
+         * consider all hosts for establishing new connections.
+         * </p>
+         *
+         * @param backgroundExecutor {@link Executor} on which to schedule health checking.
+         * @return {@code this}.
+         * @see #healthCheckFailedConnectionsThreshold(int)
+         */
+        public RoundRobinLoadBalancerFactory.Builder<ResolvedAddress, C> backgroundExecutor(
+                Executor backgroundExecutor) {
+            this.backgroundExecutor = requireNonNull(backgroundExecutor);
+            return this;
+        }
+
+        /**
+         * Configure an interval for health checking a host that failed to open connections.
+         * <p>
+         * {@link #healthCheckFailedConnectionsThreshold(int)} can be used to disable the health checking mechanism
+         * and always consider all hosts for establishing new connections.
+         * </p>
+         * @param interval interval at which a background health check will be scheduled.
+         * @return {@code this}.
+         * @see #healthCheckFailedConnectionsThreshold(int)
+         */
+        public RoundRobinLoadBalancerFactory.Builder<ResolvedAddress, C> healthCheckInterval(Duration interval) {
+            if (requireNonNull(interval).isNegative()) {
+                throw new IllegalArgumentException("Health check interval can't be negative");
+            }
+            this.healthCheckInterval = interval;
+            return this;
+        }
+
+        /**
+         * Configure a threshold for consecutive connection failures to a host. When the {@link LoadBalancer} fails
+         * to open a connection in more consecutive attempts than the specified value, the host will be marked as
+         * unhealthy and a connection establishment will take place in the background. Until finished, the host will
+         * not take part in load balancing selection.
+         * Use a negative value of the argument to disable health checking.
+         * @param threshold number of consecutive connection failures to consider a host unhealthy and eligible for
```

Comment:
What does zero mean? Does 1 mean a single failure or one additional failure following a failure?

Reply:
Thanks for this comment. Actually there were inconsistencies here. I adjusted the description (# consecutive failures greater or equal to provided value triggers health check).
I'm throwing an exception for 0 now. 1 means any failure triggers the health check. Hope it's clearer now.